ote/src/cljc/ote/transit_changes.cljc

```diff
@@ -309,8 +313,28 @@
    [25 12] :xmas-day
    [26 12] :boxing-day})
 
-(defn is-holiday?
-  "Check if given date is a holiday to be skipped. Returns holiday id if it is or nil otherwise."
-  [date]
-  (let [{::time/keys [month date]} (time/date-fields date)]
-    (static-holidays [date month])))
+#?(:clj
+   (defn exception-holidays
+    "returns a map like {[day month year] :holiday} based on detection-holidays table"
+    [db]
+    (let [dates (specql/fetch db
+                              :gtfs/detection-holidays
+                              #{:gtfs/date :gtfs/reason}
+                              {:gtfs/date op/not-null?})]
+      (reduce
+        (fn [map holiday]
+          (let [{::time/keys [month date year]} (time/date-fields (:gtfs/date holiday))]
+            (assoc map
+              [date month year] (keyword (str/lower-case (:gtfs/reason holiday))))))
```

Comment:
Was it handled somewhere if `holiday` is null, e.g. `""` in CSV? I.e. can this rely that holiday name string is guaranteed to be valid.

Reply:
yes by default it's just a string of "holiday" if the third arg from the csv is empty.